src/hooks/useSendTransaction.ts

```diff
@@ -0,0 +1,452 @@
+import { atom, useAtom } from 'jotai'
+import { ethers } from 'ethers'
+import { getNetworkNiceNameByChainId, getNetworkNameAliasByChainId } from '@pooltogether/utilities'
+
+import { useOnboard } from './useOnboard'
+
+const getRevertReason = require('eth-revert-reason')
+
+export const transactionsAtom = atom([])
+
+const DEFAULT_TRANSACTIONS_KEY = 'txs'
+
+const DEFAULT_TX_DETAILS = {
+  name: '',
+  contractAbi: [],
+  contractAddress: '',
+  method: '',
+  value: 0,
+  params: [],
+  callbacks: {}
+}
+
+/**
+ * Finds a transaction in the mem-store based on the provided txId
+ * @param txId
+ * @returns object (Transaction)
+ */
+export const useTransaction = (txId) => {
```

Comment:
nit: move this to its own `useTransaction.ts` file, name this file `transactions.ts` or even make a nested `/hooks/transactions` folder. I don't want to forget where things are